service/src/main/java/bio/terra/workspace/service/workspace/WorkspaceService.java

```diff
@@ -285,4 +296,52 @@ public String getRequiredGcpProject(UUID workspaceId) {
         .getGcpCloudContext()
         .map(GcpCloudContext::getGcpProjectId);
   }
+
+  /**
+   * Grant a user permission to impersonate their pet service account in a given workspace. Unlike
+   * other operations, this does not run in a flight because it only requires one write operation.
+   * This operation is idempotent.
+   *
+   * @return The email identifier of the user's pet SA in the given workspace.
+   */
+  public String enablePet(UUID workspaceId, AuthenticatedUserRequest userRequest) {
```

Comment:
I'd name this `enablePetServiceAccountImpersonation` or something else more complete.